packages/bsky/src/hydration/hydrator.ts

```diff
@@ -448,7 +452,14 @@ export class Hydrator {
       if (parentDid) {
         const pair: RelationshipPair = [creator, parentDid]
         relationships.push(pair)
-        postBlockPairs.reply = pair
+        postBlockPairs.parent = pair
+      }
+      const rootUri = post.reply?.root.uri
+      const rootDid = rootUri && didFromUri(rootUri)
+      if (rootDid) {
+        const pair: RelationshipPair = [creator, rootDid]
+        relationships.push(pair)
```

Comment:
Only adding `rootDid` if it's different from `parentDid` may be a nice way to mitigate unnecessary extra fetching.  We actually may even want to add a check to ensure we don't try to fetch `[creator, creator]`, which I can imagine being pretty common.

Reply:
To be more explicit about the cases I'm thinking of—
```ts
if (rootDid !== creator) {}
if (parentDid !== creator && parentDid !== rootDid) {}
```

Reply:
The logic in `replyRef` view is to check for `childBlocks?.root || parentBlocks?.root`, so not setting `root` if it's the same as `parent` would mean we'd need to check for `childBlocks.parent && parent.uri === root.uri`. We had a check like that in `replyRef` but I think the logic is a little more clear now.

`getBidirectionalBlocks` dedupes these pairs, so if `parentDid === rootDid`, I think it should be OK so long as neither `=== creator`, since in the end it will fetch the relationship once and we'll set it on both `parent` and `root`.

So I think just a check on both conditions here for `did !== creator` should serve well?